src/src_mysql/sadb_routine_mariadb.template.c

```diff
@@ -96,20 +99,111 @@ static int32_t sadb_config(void)
+                }
+            }
+        }//end if TLS connection    
+            //else regular username & password connection 
+        else {
+            //if NULL is returned then there is an error, else success
+            if (mysql_real_connect(con, sadb_mariadb_config->mysql_hostname,
+                    sadb_mariadb_config->mysql_username,
+                    sadb_mariadb_config->mysql_password,
+                    sadb_mariadb_config->mysql_database,
+                    sadb_mariadb_config->mysql_port, NULL, 0) == NULL) {
+                //0,NULL,0 are port number, unix socket, client flag
+                status = finish_with_error(con, SADB_MARIADB_CONNECTION_FAILED);
+            } else {
+                status = CRYPTO_LIB_SUCCESS;
+            }
+        }//end regular password 
     }
-
+    return status;
+}
+/*===========================================================================
+Function:           encrypted_tls_connection
+Description:        Attempts to establish a encrypted connection using TLS.  
+Inputs:             MYSQL* con, SadbMariaDBConfig_t* sadb_mariadb_config         
+Outputs:            status - int32
+References:         1) https://dev.mysql.com/doc/c-api/8.0/en/c-api-encrypted-
+ *                      connections.html#c-api-enforcing-encrypted-connection
+ *                  2) https://dev.mysql.com/doc/c-api/8.0/en/mysql-ssl-set.html
+ *                  3) https://www.xuchao.org/docs/mysql/connectors-apis.html#c-api-encrypted-connections
+Example call:        
+Note:               MySQL server MUST be configured for encrypted connections:
+ *                  https://dev.mysql.com/doc/refman/5.7/en/using-encrypted-connections.html
+==========================================================*/
+static void encrypted_tls_connection(MYSQL* con, SadbMariaDBConfig_t* sadb_mariadb_config) {
+    if (NULL != con) {
+        /*Based documentation mysql_ssl_set() always returns 0.
+         Therefore successful connections can only be checked'
+         via subsequent call to mysql_real_connect()*/
+        mysql_ssl_set(con,
+                sadb_mariadb_config->ssl_key,
+                sadb_mariadb_config->ssl_cert,
+                sadb_mariadb_config->ssl_ca,
+                sadb_mariadb_config->ssl_capath, NULL);
+    }
+}
+/*===========================================================================
+Function:           is_user_signed_into_db
+Description:        Queries mariadb for users that are signed & if the expected
+ *                  username appearss then the function return success else error. 
+Inputs:             MYSQL* con, char* username        
+Outputs:            status - int32 (success or error)
+References:         1) techonthenet.com/mysql/questions/find_users_logged_in.php
+==========================================================*/
+static int32_t is_user_signed_into_db(MYSQL* con, char* username) {
```

Comment:
As far as I can tell, this function isn't actually required/a core part of making a TLS/mTLS connection to a MySQL DB -- it's not necessary and should be deleted.

Reply:
Its removed, however, It checks whether the actual user is logged in.